pkg/sbom/node_test.go

```diff
@@ -304,3 +304,152 @@ func TestNodeCopy(t *testing.T) {
 	require.Equal(t, "f3ae11065cafc14e27a1410ae8be28e600bb8336", copied.Hashes[int32(HashAlgorithm_SHA1)])
 	require.Equal(t, "TEXT", copied.FileTypes[0])
 }
+
+func TestNodeDescendants(t *testing.T) {
+	sutId := "mynode"
+	for _, tc := range []struct {
+		name                string
+		sut                 *NodeList
+		expectedNodesLength int
+		depth               int
+	}{
+		{
+			//     mynode
+			//       |
+			name: "single node",
+			sut: &NodeList{
+				Nodes:        []*Node{{Id: sutId}},
+				Edges:        []*Edge{},
+				RootElements: []string{sutId},
+			},
+			expectedNodesLength: 1,
+			depth:               10,
```

Comment:
I don't understand why depth is expected to be 10 here? I'm definitely missing something.

Reply:
Ah this is just a "big enough number" to capture the expected (1-node) graph. It is larger than than the expected length because we want to catch any bugs in the traversal that may be causing duplicates.